src/database/database.cpp

```diff
@@ -29,434 +29,462 @@
-    // --- Reading ---
-    "markReadAfterSecondsEnable int, "    // Enable "Read" timer
-    "markReadAfterSeconds int, "          // Number of seconds that must elapse to mark news "Read"
-    "markReadInNewspaper int, "           // mark Read when Newspaper layout
-    "markDisplayedOnSwitchingFeed int, "  // mark Read on switching to another feed
-    "markDisplayedOnClosingTab int, "     // mark Read on tab closing
-    "markDisplayedOnMinimize int, "       // mark Read on minimizing to tray
-    // --- Display ---
-    "layout text, "      // news display layout
-    "filter text, "      // news display filter
-    "groupBy int, "      // column number to sort by
-    "displayNews int, "  // 0 - display content from news; 1 - download content from link
-    "displayEmbeddedImages integer default 1, "  // display images embedded in news
-    "loadTypes text, "                           // type of content to load ("images" or "images sounds" - images only or images and sound)
-    "openLinkOnEmptyContent int, "               // load link, if content is empty
-    // --- Columns ---
-    "columns text, "  // columns list and order of the news displayed in list
-    "sort text, "     // column name to sort by
-    "sortType int, "  // sort type (ascend, descend)
-    // --- Clean Up ---
-    "maximumToKeep int, "           // maximum number of news to keep
-    "maximumToKeepEnable int, "     // enable limitation
-    "maximumAgeOfNews int, "        // maximum store time of the news
-    "maximumAgoOfNewEnable int, "   // enable limitation
-    "deleteReadNews int, "          // delete read news
-    "neverDeleteUnreadNews int, "   // don't delete unread news
-    "neverDeleteStarredNews int, "  // don't delete starred news
-    "neverDeleteLabeledNews int, "  // don't delete labeled news
-    // --- Status ---
-    "status text, "                 // last update result
-    "created text, "                // feed creation timestamp
-    "updated text, "                // last update timestamp
-    "lastDisplayed text, "           // last display timestamp
-    "f_Expanded integer default 1, "  // expand folder flag
-    "flags text, "                    // more flags (example "focused", "hidden")
-    "authentication integer default 0, "    // enable authentification, sets on feed creation
-    "duplicateNewsMode integer default 0, " // news duplicates process mode
-    "typeFeed integer default 0, "          // reserved for future purposes
-    "showNotification integer default 0, "  //
-    "disableUpdate integer default 0, "     // disable update feed
-    "javaScriptEnable integer default 1, "  //
-    // version 16
-    "layoutDirection integer default 0, "    // 0 - ltr; 1 - rtl
-    // Version 17
-    "SingleClickAction integer default 0, " // ENewsClickAction
-    "DoubleClickAction integer default 0, " // ENewsClickAction
-    "MiddleClickAction integer default 0 "  // ENewsClickAction
-    ")");
+        "CREATE TABLE feeds("
```

Comment:
More needless whitespace changes.

Reply:
Ok. In your opinion, what's the best way to fix this automatic unwanted changes?

Reply:
I would use a Git frontend (I prefer GNOME Gitg) combined with the console and follow these steps (roughly):
1. Create a backup branch (`git branch backup-branch`)
2. `git reset HEAD~5` to unpush the latest changes from your git history. File changes are still preserved.
3. use the git gui (or command line tool) to revert the needless whitespace and encoding changes. Command line: `git checkout -- [filename]`
4. Copy commit messages from your "old" commits

For the future, I would recommend disabling auto-formatting the code in your editor/IDE. Yeah, that's quite annoying.

Reply:
With `git push --force` you should be able to push the modified changes back to this pull request.

Reply:
I like the idea of this pull request, it is just hard to have a look at the code if the diff is cluttered by so much noise.

Reply:
Thank you very much.
It's my first time to share a code in github, and I don't expect this issues.
I will try to fix this ASAP.

Reply:
Feel free to ask for more help and welcome to the contribution world ;)